frontend/src/app/services/task.service.ts

```diff
@@ -23,29 +23,42 @@ interface LabelTagResponse {
 @Injectable()
 export class TaskService {
 
-    private currentTask: Task;
+    private tasks: Array<Task> = [];
+    constructor(private http: HttpClient) { }
 
-    constructor(private http: HttpClient) {}
-
-    setCurrentTask(task: Task): void {
-        this.currentTask = task;
-    }
-
-    getCurrentTask(): Task {
-        return this.currentTask;
+    getCurrentTask(taskKey: string): Promise<Task> {
+        if (this.tasks.length > 0) {
```

Comment:
I would prefer `taskKey` to be null/undefined rather than empty string. It would be more readable and easier to think about.

What's more, please leave a comment somewhere in the code that there are two possibilities to get current Task - from URL and from Tasks Page. I bet that after a few months we won't remember about it...